org.scala.tools.eclipse.search/src/org/scala/tools/eclipse/search/handlers/OpenCallHierarchy.scala

```diff
@@ -0,0 +1,91 @@
+import org.scala.tools.eclipse.search.ui.CallHierarchyView
+import org.scala.tools.eclipse.search.ui.Node
+import org.eclipse.jface.operation.IRunnableWithProgress
+import org.eclipse.jface.dialogs.ProgressMonitorDialog
+import scala.tools.nsc.interactive.Response
+import scala.reflect.internal.util.OffsetPosition
+import org.scala.tools.eclipse.search.ui.CallerNode
+
+class OpenCallHierarchy extends AbstractHandler with HasLogger {
+  private val finder: Finder = SearchPlugin.finder
+  private val reporter: ErrorReporter = new DialogErrorReporter
+
+  override def execute(event: ExecutionEvent): Object = {
+
+    def scheduleJob(e: Entity, scope: Scope, in: Node) = {
+      val job = new Job("Searching for callers...") {
+        def run(monitor: IProgressMonitor): IStatus = {
+
+          finder.findCallers(e, scope, (offset, ee, label, owner, project) => in.list = CallerNode(offset, ee, scope, label, owner, project) :: in.list, monitor)
+          Status.OK_STATUS
+        }
+      }
+      job.schedule()
+
+    }
+
+    val view = PlatformUI.getWorkbench()
+      .getActiveWorkbenchWindow()
+      .getActivePage()
+      .showView(CallHierarchyView.VIEW_ID)
+
+    for {
+      // For the selection
+      editor <- Option(HandlerUtil.getActiveEditor(event)) onEmpty reporter.reportError("An editor has to be active")
+      scalaEditor <- editor.asInstanceOfOpt[InteractiveCompilationUnitEditor] onEmpty reporter.reportError("Active editor wasn't a Scala editor")
+      selection <- UIUtil.getSelection(scalaEditor) onEmpty reporter.reportError("You need to have a selection")
+      thview <- view.asInstanceOfOpt[CallHierarchyView] onEmpty logger.debug("Wasn't an instance of CallHierarchyView")
+    } {
+      // Get the relevant scope to search for sub-types in.
+      val projects = ProjectFinder.projectClosure(scalaEditor.getInteractiveCompilationUnit.scalaProject.underlying)
+      val scope = Scope(projects.map(IScalaPlugin().asScalaProject(_)).flatten)
+      val loc = Location(scalaEditor.getInteractiveCompilationUnit, selection.getOffset())
+
+      val e = finder.entityAt(loc)
+      val root = thview.input
+      root.list = List()
+      e match {
+        case Right(Some(entity: Method)) =>
+          scheduleJob(entity, scope, root)
+        case Right(Some(entity: Val)) =>
+          scheduleJob(entity, scope, root)
+        case Right(Some(entity: Var)) =>
+          scheduleJob(entity, scope, root)
+        case Right(Some(entity)) => reporter.reportError(s"Sorry, can't use selected '${entity.name}' to build a call-hierarchy.")
+        case Right(None) => // No-op
+        case Left(_) =>
+          reporter.reportError("Sorry, couldn't get the symbol of the given entity.\n\n" +
+            "This is very likely a bug, so please submit a bug report that contains\n" +
+            "a minimal example to https://www.assembla.com/spaces/scala-ide/tickets\n\n" +
+            "Thank you! - IDE Team")
```

Comment:
I don't like when a computer says it is sorry - they aren't actually. Also, you could have used multi line strings + strip margins here to make the string more readable.